libraries/voicerecorder/impl/src/main/kotlin/io/element/android/libraries/voicerecorder/impl/VoiceRecorderImpl.kt

```diff
@@ -67,6 +69,7 @@ class VoiceRecorderImpl @Inject constructor(
     private var audioReader: AudioReader? = null
     private var recordingJob: Job? = null
     private val levels: MutableList<Float> = mutableListOf()
+    private val lock = Mutex()
```

Comment:
Was there some concurrency thing that led to crashes before?

Reply:
Yes a concurrency modification exception when calling toPersisentList() at a later point, as I was returning a list that was being modified, so some safety measures.